### ISSUE TITLE
Packages page view mode

### ISSUE THREAD

**Issue by MiguelSOliveira** [DinoDevs/GladiatusCrazyAddon#309]
Would be cool if we could switch view modes in packages page to show as a list, instead of a grid.
This way we could see the items' names instead of their picture, which i think would be much more useful.

**Comment by FrutyX** [comment]
Hm, making an independent list would be better, but that would require GCA to crawl the packages, which is server heavy, bad idea.

So what is left is editing the current package page, removing the grinds, item images, but you still have to get the description from the item tooltips, messy.

We'll see, Gram and Apo are good at developing things, it's all about getting the right idea.

**Comment by MiguelSOliveira** [comment]
it would not require GCA to crawl the packages any more than it currently does, from what I can see. Gladiatus already returns the tooltip data in the page's html.

Something like this:
```
    <div data-no-combine="true" data-no-destack="true" data-container-number="-1061422"><div style="left:0px;top:0px" class="item-i-9-7 ui-draggable ui-droppable" data-content-type="1024" data-content-size="1" data-enchant-type="null" data-price-gold="25867" data-tooltip="[[[&quot;Manius Star talisman of Stars&quot;,&quot;lime; text-shadow: 0 0 2px #000, 0 0 2px lime&quot;],[&quot;Damage +1&quot;,&quot;#DDD&quot;],[&quot;Charisma +25&quot;,&quot;#DDD&quot;],[&quot;Charisma +2% (+2)&quot;,&quot;#DDD&quot;],[&quot;Intelligence +25&quot;,&quot;#DDD&quot;],[&quot;Critical attack value +25&quot;,&quot;#DDD&quot;],[&quot;Level 96&quot;,&quot;#808080&quot;],[&quot;Value 25.867 &lt;div class=\&quot;icon_gold\&quot;&gt;&lt;\/div&gt;&quot;,&quot;#DDDDDD&quot;],[&quot;Durability 23040\/23040 (100%)&quot;,&quot;#00ff00&quot;],[&quot;Conditioning 0\/5760 (0%)&quot;,&quot;#808080&quot;]]]" data-comparison-tooltip="null" data-level="96" data-hash="f40-9-7-jyj-0-3g-i-0-0-0-0-0-0-0-0-hs0-0-0" data-amount="1" data-position-x="1" data-position-y="1" data-measurement-x="1" data-measurement-y="1" data-basis="9-7"></div></div>
```

The attribute "data-tooltip", should give GCA what it needs, unless I'm mistaken.

But I'm sure the devs have a better understanding of what it would entail..

**Comment by MiguelSOliveira** [comment]
Just as a proof of concept, I hacked some JS to show that it's easily done.

```
document.querySelectorAll('[class^=item-]').forEach(item => {
    const tooltipData = item.getAttribute('data-tooltip')
    
    if (tooltipData !== null) {
        console.log(JSON.parse(tooltipData)[0][0][0])
    }
})
```

The following outputs the packages by name:
<img width="464" alt="imagem" src="https://user-images.githubusercontent.com/8149533/147420253-dd4d57f0-a2a1-4ad8-b37c-8a30399690bd.png">

**Comment by GramThanos** [comment]
What is the purpose of this? Why do you need a list of the items names?

Describe what exactly you want to achieve so that we can understand what exactly is needed.

**Comment by MiguelSOliveira** [comment]
For example, when I filter packages by "Scrolls" I get this:
<img width="839" alt="imagem" src="https://user-images.githubusercontent.com/8149533/147423706-691b6b9e-0798-4c9a-b134-71406fe0a50f.png">

From the images, I have no idea what each scroll actually is. It provides no information.
I think if instead of images, we had a list with the actual names we could see the results much better.

Does it make sense?

**Comment by FrutyX** [comment]
Well, I don't think we need to make a whole list, we can just use the packages the way they already are and add the name from a tooltip somewhere above or under the item image, it will need some playing with CSS (font size, color...) but it should do the trick.

I get that there isn't much space to add it, especially for long item names, it will need some playing with it.

**Comment by FrutyX** [comment]
As an example (for Gram):

![example](https://user-images.githubusercontent.com/23278552/147424250-c5b1395d-ab91-42ea-96b8-c8d6c0ccc869.PNG)

![example2](https://user-images.githubusercontent.com/23278552/147424221-d5f69739-accb-473a-91e9-78d0d1e9d8ba.PNG)

But it's just an idea.

**Comment by GreatApo** [comment]
Current list view implementation:
![image](https://user-images.githubusercontent.com/9620510/178859947-3037be5b-1104-4a97-addf-7fd28ef9cc79.png)

And with small items:
![image](https://user-images.githubusercontent.com/9620510/178860032-171bc2c6-0542-4e23-baf7-5ad428098066.png)

**Comment by SleepingShadow-bit** [comment]
Is it correct that the "Improve items layout" setting must be activated for the "list view"?

Without "improve items layout"
![IMG_20220714_062156](https://user-images.githubusercontent.com/8767665/178898373-c118c730-eaec-4eea-8b7c-beccb952b2ca.jpg)

It's a yes.  Is there some kind of setting protection (note) or should we just write it explicitly in the documentation?

**Comment by GreatApo** [comment]
> Is it correct that the "Improve items layout" setting must be activated for the "list view"?
> 
> Without "improve items layout"
> 

Oups... I need to fix this

**Comment by SleepingShadow-bit** [comment]
Can somebody add the de translation?

settings : {
	category_packages$list_view_layout : "Zeige Pakete als Listen-Ansicht",
}

**Comment by SleepingShadow-bit** [comment]
> > Is it correct that the "Improve items layout" setting must be activated for the "list view"?
> > Without "improve items layout"
> 
> Oups... I need to fix this

Hmm. I am Online via smartphone. Do u have the same bug on pc? 
![IMG_20220714_230613](https://user-images.githubusercontent.com/8767665/179086250-5a9b0c0f-2dbe-4c9b-95e4-b446935c28cc.jpg)

**Comment by GreatApo** [comment]
>Hmm. I am Online via smartphone. Do u have the same bug on pc? 
> 

I think I fixed that by merging some feature in the compact view. Do you have the last version on you phone? (You will see the layout view option as a drop down)

**Comment by SleepingShadow-bit** [comment]
Yes. I have the drop down menu (really nice).

**Comment by SleepingShadow-bit** [comment]
Same on PC
![Screenshot_2](https://user-images.githubusercontent.com/8767665/179196067-f29fa12a-a5ad-4b9f-bcf8-fac12e6d21ff.png)

**Comment by FrutyX** [comment]
I believe I am seeing the same.

**Comment by FrutyX** [comment]
I think the various item heights are causing the problems, it's a big mess... However enabling the small items size feature does fix it completely, but the big ones are problematic.

**Comment by FrutyX** [comment]
@SleepingShadow-bit can you get the latest commit and test it now? It's experimental, I can see more adjustments are needed.

**Comment by GreatApo** [comment]
@FrutyX please name the commits appropriately (you can steel keep them funny). That is helpful if we need to track a change at a later stage :)

I thought I had these fixed :/

**Comment by FrutyX** [comment]
Can't help myself, but the font-size seems just extremely small, wouldn't be at least 10px better?

![font](https://user-images.githubusercontent.com/23278552/179352321-2b3d8372-f5d9-473a-b8f7-36f13af0a500.PNG)

**Comment by GreatApo** [comment]
> Can't help myself, but the font-size seems just extremely small, wouldn't be at least 10px better?
> 
> ![font](https://user-images.githubusercontent.com/23278552/179352321-2b3d8372-f5d9-473a-b8f7-36f13af0a500.PNG)

I agree but we need to make sure it fits

**Comment by FrutyX** [comment]
10px seems to be fine, anything above that - problematic, unless more changes.

**Comment by FrutyX** [comment]
I have found a problem, it seems like a page loading problem, sometimes it gets loaded badly, I don't think it's related to the changes I made today, a page refresh does fix it, it can occur sometimes.

![one](https://user-images.githubusercontent.com/23278552/179353696-35145061-e77a-4806-a606-7115c5e77fd1.PNG)
![two](https://user-images.githubusercontent.com/23278552/179353699-92eb79be-82bd-4032-bc91-262be8a24bbd.PNG)

**Comment by GreatApo** [comment]
> I have found a problem, it seems like a page loading problem, sometimes it gets loaded badly, I don't think it's related to the changes I made today, a page refresh does fix it, it can occur sometimes.
> 

That may be due to compact style code running before the list changes...

**Comment by FrutyX** [comment]
Or perhaps I did really caused it, but in my defense - I said it was experimental :D.

**Comment by FrutyX** [comment]
> 10px seems to be fine, anything above that - problematic, unless more changes.

And scratch that too, small items (like amulets) are having problems with 10px too, not enough space.

**Comment by FrutyX** [comment]
Alright, it's probably half and a half, Apo is probably right too, with the last commit from me, it's loading fine, but I told GCA to load 100 pages,  it did occur but not even every time, only sometimes and it seems scrolling too fast is also helping to recreate it, so it is related to page loading.

I will not touch this anymore, other than that I think it's fine, but god knows what Sleeping will find again :D.

**Comment by SleepingShadow-bit** [comment]
Guys it's look good =)
Thank u for the fix =)

**Comment by SleepingShadow-bit** [comment]
Ohoo now you have also implemented my wish =) - - >  #354

**Comment by FrutyX** [comment]
Hmm, I do not have such a problem, did you tried a simple page refresh? I know this feature is problematic during the page load, but like I said, mine is fine.

Console is clear I suppose?

**Comment by SleepingShadow-bit** [comment]
am confused now Tried it several times. And in the meantime cleared cache and data. Now everything ok!

**Comment by SleepingShadow-bit** [comment]
Could it be that this is currently a cache and load issue?  The error as shown above keeps coming back.  And when removing packages, it doesn't update automatically, so I have to reload the page manually.  Am I the only one with this issue?

I will test more and more

**Comment by FrutyX** [comment]
I will add a bug label to it, I know there are problems, most importantly I want the boys to focus on the broken features first, we will see about this one.

**Comment by FrutyX** [comment]
I've just met a Gladiatus bug - the whole first page did not loaded, there was only text saying "no packages", I have disabled all extensions, and it was Gladiatus.

Hard to work on something, when the game in it's code doesn't work as it should....

**Comment by FrutyX** [comment]
@GramThanos Your code analyzing skills may be also needed hero, I only want you to check this, and if something is bad, to do code improvements to prevent the page, to be badly loaded.

However like  I said, for example on my merged server, I often have server lags, timeouts, and that is affecting the packages a lot (items not loading or not loading in time).

**Comment by GramThanos** [comment]
I don't see any problem on this.

**Comment by FrutyX** [comment]
Perfect, report here any bugs if anyone see some.

If you see this, it's only the game loading your items, and sometimes it can take a while, or does not load at all. Not an GCA bug.

![loadingpackages](https://user-images.githubusercontent.com/23278552/183218633-0ef1e0f6-5738-495a-8893-e7999feff716.PNG)

**Comment by GramThanos** [comment]
![image](https://user-images.githubusercontent.com/14858959/183239017-d6e33d35-43c2-4dc8-b700-3bd37c212dd6.png)

**Comment by FrutyX** [comment]
Looking good.

**Comment by FrutyX** [comment]
Implemented in v4.3.6, closing, re-open if needed.
